mifosng-provider/src/main/java/org/mifosplatform/portfolio/loanproduct/domain/LoanProductConfigurableAttributes.java

```diff
@@ -0,0 +1,165 @@
+public class LoanProductConfigurableAttributes extends AbstractPersistable<Long> {
+
+    @ManyToOne
+    @JoinColumn(name = "loan_product_id", nullable = true)
+    private LoanProduct loanProduct;
+
+    @Column(name = "amortization", nullable = true)
+    private Boolean configureAmortization;
+
+    @Column(name = "interest_method", nullable = true)
+    private Boolean configureInterestMethod;
+
+    @Column(name = "repayment_strategy", nullable = true)
+    private Boolean configureRepaymentStrategy;
+
+    @Column(name = "interest_calculation_period", nullable = true)
+    private Boolean configureInterestCalcPeriod;
+
+    @Column(name = "arrears_tolerance", nullable = true)
+    private Boolean configureArrearsTolerance;
+
+    @Column(name = "repay_every", nullable = true)
+    private Boolean configureRepaymentFrequency;
+    
+    @Column(name = "moratorium", nullable = true)
+    private Boolean configureMoratorium;
+    
+    @Column(name = "grace_on_arrears_aging", nullable = true)
+    private Boolean configureGraceOnArrearsAging;
+    
+    private static String[] loanConfigurableAttributes = {"configureAmortization","configureInterestMethod","configureRepaymentStrategy",
+			"configureInterestCalcPeriod","configureArrearsTolerance","configureRepaymentFrequency","configureMoratorium","configureGraceOnArrearsAging"};
+    
+	public static LoanProductConfigurableAttributes createFrom(
+			JsonArray attributeArray) {
+
+		final Boolean[] values = new Boolean[8];
+		
+			for (int i = 0; i < values.length; i++) {
+				JsonElement attributeElement = attributeArray.get(i);
+				if (attributeElement.isJsonObject()) {
+					final JsonObject object = attributeElement.getAsJsonObject();
+
+				if (object.has(getAllowedLoanConfigurableAttributes()[i])
+						) {
+					values[i] = object.get(
+							getAllowedLoanConfigurableAttributes()[i])
+							.getAsJsonPrimitive().getAsBoolean();
+				}
+			}
+		}
+		final Boolean configureAmortization = values[0];
+		final Boolean configureInterestMethod = values[1];
+		final Boolean configureRepaymentStrategy = values[2];
+		final Boolean configureInterestCalcPeriod = values[3];
+		final Boolean configureArrearsTolerance = values[4];
+		final Boolean configureRepaymentFrequency = values[5];
+		final Boolean configureMoratorium = values[6];
+		final Boolean configureGraceOnArrearsAging = values[7];
+
+		return new LoanProductConfigurableAttributes(configureAmortization,
+				configureInterestMethod, configureRepaymentStrategy,
+				configureInterestCalcPeriod, configureArrearsTolerance,
+				configureRepaymentFrequency, configureMoratorium,
+				configureGraceOnArrearsAging);
+	}
+		
+	 public void updateLoanProduct(final LoanProduct loanProduct) {
+	        this.loanProduct = loanProduct;
+	    }
+          
+	 public static LoanProductConfigurableAttributes populateDefaultsForConfigurableAttributes(){
+         final Boolean configureAmortization = false;
```

Comment:
there are two scenarios here

1) For a user who does not use these optional paramters, the API should behave exactly the same way as it did earlier

2) From the community app, you always use the new options , so are free to default it to whatever you want